hles/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/DietTransformations.scala

```diff
@@ -156,77 +156,96 @@ object DietTransformations {
     )
   }
 
-  /** Map fields about supplements taken daily by a dog. */
-  def mapDailySupplements(rawRecord: RawRecord, dog: HlesDogDiet): HlesDogDiet = {
-    val otherSupps = rawRecord.getOptionalNumber("df_s_other").filter(_ != 0L)
-
-    dog.copy(
-      dfDailySupplementsBoneMeal = rawRecord.getOptionalNumber("df_s_bone_meal").filter(_ != 0L),
-      dfDailySupplementsGlucosamine =
-        rawRecord.getOptionalNumber("df_s_glucosamine").filter(_ != 0L),
-      dfDailySupplementsChondroitin =
-        rawRecord.getOptionalNumber("df_s_chondroitin").filter(_ != 0L),
-      dfDailySupplementsOtherJoint =
-        rawRecord.getOptionalNumber("df_s_joint_other").filter(_ != 0L),
-      dfDailySupplementsOmega3 = rawRecord.getOptionalNumber("df_s_omega3").filter(_ != 0L),
-      dfDailySupplementsNonOilSkin = rawRecord.getOptionalNumber("df_s_skin").filter(_ != 0L),
-      dfDailySupplementsVitamins = rawRecord.getOptionalNumber("df_s_vitamin").filter(_ != 0L),
-      dfDailySupplementsEnzyme = rawRecord.getOptionalNumber("df_s_enzyme").filter(_ != 0L),
-      dfDailySupplementsProbiotics = rawRecord.getOptionalNumber("df_s_probiotics").filter(_ != 0L),
-      dfDailySupplementsFiber = rawRecord.getOptionalNumber("df_s_fiber").filter(_ != 0L),
-      dfDailySupplementsAlkalinize = rawRecord.getOptionalNumber("df_s_alkalinize").filter(_ != 0L),
-      dfDailySupplementsAcidify = rawRecord.getOptionalNumber("df_s_acidify").filter(_ != 0L),
-      dfDailySupplementsTaurine = rawRecord.getOptionalNumber("df_s_taurine").filter(_ != 0L),
-      dfDailySupplementsAntiox = rawRecord.getOptionalNumber("df_s_antiox").filter(_ != 0L),
-      dfDailySupplementsCoenzymeQ10 = rawRecord.getOptionalNumber("df_s_q10").filter(_ != 0L),
-      dfDailySupplementsOther = otherSupps,
-      dfDailySupplementsOtherDescription = if (otherSupps.nonEmpty) {
-        rawRecord.getOptional("df_s_other_text")
+  /** Map fields about supplements taken daily by a dog when available. */
+  def mapDailySupplements(rawRecord: RawRecord, dog: HlesDogDiet): HlesDogDiet =
+    rawRecord.getOptionalBoolean("df_supplement_daily").fold(dog) { dailySupps =>
```

Comment:
we can probably turn this into a proper `if/else` statement instead of wrapping it in the `fold` by using `getRequired` instead of `getOptionalBoolean` here, since this field is required.